src/components/TextInput/BaseTextInput.js

```diff
@@ -388,7 +403,7 @@ class BaseTextInput extends Component {
                         ]}
                         onLayout={(e) => this.setState({textInputWidth: e.nativeEvent.layout.width + 2, textInputHeight: e.nativeEvent.layout.height})}
```

Comment:
I was thinking that this could be a could point to set the `state.value = defaultHiddenInputValue` when `this.props.shouldWaitWidthCalculation = true`, so you don't have to wait for another cycle to update `state.value`.

Reply:
No, In the same first parameter. I think that may be the earliest time we can set the `state.value` without running into the width problem cutting off the number.